Update roborock documentation to incorporate guidance in the quality scale (#37929)
home-assistant/home-assistant.github.io: source/_integrations/roborock.markdown

ha_integration_type: integration
---

> The Roborock {% term integration %} allows you to connect your [Roborock](https://us.roborock.com/pages/robot-vacuum-cleaner) robotic vacuums to your Home Assistant. Roborock vacuums are
> intelligent home cleaning robots and depending on the specific device may have features
> like mopping capabilities, laser navigation, and options for changing cleaning
> performance or location in the home. This integration enables you to control and
> monitor your Roborock vacuum directly from Home Assistant.

Reviewer: ```suggestion
The Roborock {% term integration %} allows you to connect your [Roborock](https://us.roborock.com/pages/robot-vacuum-cleaner) robotic vacuums to your Home Assistant. Roborock vacuums are
intelligent home cleaning robots and, depending on the specific device, may have features
like mopping capabilities, laser navigation, and options for changing cleaning
performance or location in the home. This integration enables you to control and
monitor your Roborock vacuum directly from Home Assistant.
```

The integration also allows for automation and integration with other smart home
devices. For example, you could send a notification when the vacuum is stuck, or
pause the vacuum when a media player starts playing music.

## Prerequisites

1. Download the Roborock App for iOS or Android.

> 1. Create an account and login
> 1. Add your Roborock device to the Roborock App (for example, by scanning a QR code)

Reviewer: ```suggestion
2. Create an account and log in.
3. Add your Roborock device to the Roborock App (for example, by scanning a QR code).
```

- **Mop intensity**
  - **Description**: How hard you would like your vacuum to mop.

- **Selected map**
  - **Description**: Choose the map that is loaded on the vacuum.

- **Empty mode**
  - **Description**: You can set the "empty mode" setting including Max, Light, Balanced, and Smart.
  - **Availability**: For vacuum equipped with an auto-empty dock

#### Binary sensor

- **Charging**
  - **Description**: States if the vacuum is currently charging or not.

- **Cleaning**
  - **Description**: States if the vacuum is currently cleaning or not. This is on when the robot is actively moving around or when the robot returns to the dock when the battery is low but a clean is still active and will resume later.

- **Mop attached**
  - **Description**: States if the mop is currently attached.

- **Mop drying status**
  - **Description**: Only available on docks with drying capabilites - States if the mop is currently being driven.

- **Water box attached**
  - **Description**: States if the water box is currently attached.

- **Water shortage**
  - **Description**: States if the water box is low on water - 'Ok' if it has not detected a water shortage.

#### Sensor

- **Cleaning area**

>   - **Description**: How much area the vacuum has cleaned in its current run.  If the vacuum is not currently cleaning, how much area it has cleaned during its last run.

Reviewer: ```suggestion
  - **Description**: How much area the vacuum has cleaned in its current run. If the vacuum is not currently cleaning, how much area it has cleaned during its last run.
```

- **Cleaning time**
  - **Description**: How long the vacuum has been cleaning for. If the vacuum is not currently cleaning, how long it cleaned for in its last run.

- **Cleaning progress**
  - **Description**: Only available on some newer devices - what percent of the current cleaning is completed.

- **Dock error**

>   - **Description**: Only available on the non-basic docks - The current error of the vacuum or 'Ok' if none exist

Reviewer: ```suggestion
  - **Description**: Only available on the non-basic docks - The current error of the vacuum or 'Ok' if none exists.
```

#### Number

- **Volume**
  - **Description**: This allows you to control the volume of the robot's voice. For example, when it states "Starting cleaning". This allows you to set the volume to 0%, while the app limits it to 20%.

#### Button

There are currently four buttons that allow you to reset the various maintenance items on your vacuum. Pressing the button cannot be undone. For this reason, the buttons are disabled by default to make sure they are not pressed unintentionally.

- **Reset sensor consumable**
  - **Description**: The sensors on your vacuum are expected to be cleaned after 30 hours of use.

- **Reset side brush consumable**
  - **Description**: The side brush is expected to be replaced every 200 hours.

- **Reset main brush consumable**
  - **Description**: The main brush/ roller is expected to be replaced every 300 hours.

- **Reset air filter**
  - **Description**: The air filter is expected to be replaced every 150 hours.

#### Actions

##### Action Set Vacuum Goto Position

The `roborock.set_vacuum_goto_position` action will set the vacuum to go to
the specified coordinates.

> | Data attribute   | Optional | Description                                          |

Reviewer: so, about this... I like the original version here better. I know the example in the dev docs uses a table. Looking at this, I think the example should be changed. Can you revert this table and the one below?

Reviewer: https://github.com/home-assistant/developers.home-assistant/pull/2602

Author: Done.

### How can I clean a specific room?
We plan to make the process simpler in the future, but for now, it is a multi-step process.
1. Make sure to first name the rooms in the Roborock app; otherwise, they won't appear in the debug log.

> 1. Go to {% my developer_call_service service="roborock.get_maps" title="**Developer Tools** > **Actions** > **Roborock: Get Maps**" %}. Select your vacuum as the entity. Note that room IDs and names are only updated on the currently selected map.

Reviewer: ```suggestion
2. Go to {% my developer_call_service service="roborock.get_maps" title="**Developer Tools** > **Actions** > **Roborock: Get Maps**" %}. Select your vacuum as the entity. Note that room IDs and names are only updated on the currently selected map.
```

              "17": Living room
      ```

> 1. Go back to {% my developer_call_service service="vacuum.send_command" title="**Developer Tools** > **Actions** > **Vacuum: Send Command**" %} then type `app_segment_clean` as your command and `segments` with a list of the 2-digit IDs you want to clean. Then, add `repeat` with a number (ranging from 1 to 3) to determine how many times you want to clean these areas.

Reviewer: ```suggestion
3. Go back to {% my developer_call_service service="vacuum.send_command" title="**Developer Tools** > **Actions** > **Vacuum: Send Command**" %} then type `app_segment_clean` as your command and `segments` with a list of the 2-digit IDs you want to clean. Then, add `repeat` with a number (ranging from 1 to 3) to determine how many times you want to clean these areas.
```

Roborock servers require accepting a user agreement before using the API, which may block Home Assistant during setup. Additionally, the Roborock may ask you to re-enter the user agreement, even if you have entered it before.  To allow Home Assistant to use the Roborock API, you need to take the following steps:
1. Open your Roborock app.

> 1. Open **Profile** > **About Us** > **User Agreement & Privacy Policy**.
> 1. Hit **Revoke authorization**.
> 1. Log back in and accept the policy.
> 1. Reload the Roborock integration!

Reviewer: ```suggestion
2. Open **Profile** > **About Us** > **User Agreement & Privacy Policy**.
3. Hit **Revoke authorization**.
4. Log back in and accept the policy.
5. Reload the Roborock integration!
```
did you see this numbering style in the examples somewhere? If so, the example should be fixed. I think the style is still "numbers increasing".
